packages/base/Collapse/src/Collapse/Collapse.tsx

```diff
@@ -0,0 +1,114 @@
+import React, { useRef, useState, forwardRef } from 'react'
+import { Transition } from 'react-transition-group'
+import type { BaseProps, TransitionProps } from '@toptal/picasso-shared'
+import { useMultipleForwardRefs } from '@toptal/picasso-utils'
+import cx from 'classnames'
+import { twMerge } from 'tailwind-merge'
+
+export interface Props extends TransitionProps, BaseProps {
+  /* Element that accepts ref */
+  children: React.ReactElement
+  /* Show the component; triggers the enter or exit states */
+  in?: boolean
+  /* Trigger the transition on the first mount, regardless of the `in` prop. */
+  appear?: boolean
+  /* Unmount the component on exit */
+  unmountOnExit?: boolean
+  /* Callback fired when the component has entered */
+  onEnter?: (node: HTMLElement, isAppearing: boolean) => void
+}
+
+export const Collapse = forwardRef<HTMLDivElement, Props>(
+  (
+    {
+      children,
+      className,
+      in: inProps = false,
+      timeout = 350,
+      unmountOnExit,
+      style,
+      appear,
+      'data-testid': dataTestId,
+      ...rest
+    },
+    ref
+  ) => {
+    const collapseRef = useRef<HTMLDivElement>(null)
+    const combinedRef = useMultipleForwardRefs([ref, collapseRef])
+
+    const [height, setHeight] = useState<number>()
+
+    const onEntering = () => {
+      // onShow && onShow()
+      if (collapseRef.current) {
+        setHeight(collapseRef.current.scrollHeight)
+      }
+    }
+
+    const onEntered = () => {
+      // setHeight(0)
+    }
+
+    const onExit = () => {
+      if (collapseRef.current) {
+        setHeight(collapseRef.current.scrollHeight)
+      }
+    }
+
+    const onExiting = () => {
+      // onHide && onHide()
+
+      setHeight(0)
+    }
```

Comment:
I think that maybe we can remove the unnecessary code, the comments can be used for documentation, but I think we can simplify the code by removing the `no-op` parts. If not needed maybe we can even remove the `onEntered` function.
```suggestion
    const onEntering = () => {
      if (collapseRef.current) {
        setHeight(collapseRef.current.scrollHeight)
      }
    }

    const onExit = () => {
      if (collapseRef.current) {
        setHeight(collapseRef.current.scrollHeight)
      }
    }

    const onExiting = () => {
      setHeight(0)
    }
```

Reply:
Good point. I think I'll just remove those, can be added later if we need them